formal-spec/candidateproperties.tex

```diff
@@ -0,0 +1,153 @@
+\section{Candidate Properties}
+
+\subsection{Global Properties}
+
+\begin{property}[Software Update Proposal]\label{prop:submissions-for-all}
+Any stakeholder will always be able to propose a software update to be voted by the stakeholders' community and this will be recorded as an on-chain immutable event on the blockchain.
+\end{property}
+
+\begin{property}[Non Approved Software Updates]\label{prop:nonapproved-sus}
+A software update that is not approved by the stakeholders' community will never be applied,
+\end{property}
+
+\begin{property}[Approved Software Updates]\label{prop:approved-sus}
+A software update that has been approved by the stakeholders' community will be eventually applied.
+\end{property}
+
+\begin{property}[Security of Updated Ledger]\label{prop:updated-ledger-security}
+Upon activation of the changes of an approved software update for the updated ledger properties of \emph{persistence} and \emph{liveness} will hold.
+\end{property}
+
+\begin{property}[Conflict Resolution]\label{prop:conflict-res}
+Two or more software updates that have conflicting changes will not be applied both, but a conflict resolution will take place.
+\end{property}
+
+\begin{property}[Update Dependencies]\label{prop:update-dep}
+A software update will never be applied if its dependency requirements are not met
+\end{property}
+
+\begin{property}[Duration of Voting Period]\label{prop:dur-vot-period}
+The period required for approving a software update is not fixed but can be extended by the stakeholders' community.
+\end{property}
+
+\subsection{Properties based on Non-Functional Requirements}
+
+\begin{property}[Transaction Throughput]\label{prop:tr-throughput}
+The ledger transaction throughput is not significantly impacted by the presence of software updates.
+\end{property}
+
+\begin{property}[Blockchain Size]\label{prop:blockchain-size}
+The blockchain size is not significantly impacted by the presence of software updates.
+\end{property}
+
+\subsection{Consensus Protocol Properties}
+
+\begin{property}[Bootstrapping]\label{prop:bootstrap}
+The update mechanism leverages the consensus protocol of the blockchain itself.
+\end{property}
+
+\begin{property}[Resiliency]\label{prop:resiliency}
+The resiliency of the consensus protocol (expressed as the fraction ($t/n$ where $t$ is the adversary stake and $n$ is the total stake) of misbehaving parties a protocol can tolerate) is not affected by the update mechanism
```

Comment:
We can't prevent that the software gets updated to something that isn't resilient. Or are you referring to the resiliency of the consensus protocol prior to the update?